snapcraft/cli/_errors.py

```diff
@@ -124,6 +125,11 @@ def exception_handler(
             traceback.print_exception(*exc_info)
             click.echo(_MSG_MANUALLY_REPORT)
         else:
+            if is_connected_to_tty:
+                click.echo(_MSG_TRACEBACK_FILE)
+            else:
+                click.echo(_MSG_TRACEBACK_PRINT)
+                traceback.print_exception(*exc_info)
```

Comment:
this scenario of no tty but with input looks weird, the yes/no prompt should be the bottom most message as I am still expected to fill in this data, we are writing this to a file (maybe either or but not both).
```
sergiusens@tupa:~/source/snapcraft$ ssh ubuntu@10.76.47.78 "cd source/make-hello && PATH=/home/ubuntu/venv/snapcraft/bin:/bin:/usr/bin snapcraft"
Pulling make-project 
Sorry, an error occurred in Snapcraft:
We would appreciate it if you anonymously reported this issue.
No other data than the traceback ('/tmp/tmp6l6mf4oy/trace.txt') and the version of snapcraft in use will be sent.
Would you like to send this error data? (Yes/No/Always) [no]: Traceback (most recent call last):
  File "/home/ubuntu/venv/snapcraft/bin/snapcraft", line 11, in <module>
    load_entry_point('snapcraft==2.43.1', 'console_scripts', 'snapcraft')()
  File "/home/ubuntu/venv/snapcraft/lib/python3.5/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/ubuntu/venv/snapcraft/lib/python3.5/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/ubuntu/venv/snapcraft/lib/python3.5/site-packages/click/core.py", line 1043, in invoke
    return Command.invoke(self, ctx)
  File "/home/ubuntu/venv/snapcraft/lib/python3.5/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ubuntu/venv/snapcraft/lib/python3.5/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/ubuntu/venv/snapcraft/lib/python3.5/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/ubuntu/venv/snapcraft/lib/python3.5/site-packages/snapcraft/cli/_runner.py", line 93, in run
    ctx.forward(lifecyclecli.commands["snap"])
  File "/home/ubuntu/venv/snapcraft/lib/python3.5/site-packages/click/core.py", line 553, in forward
    return self.invoke(cmd, **kwargs)
  File "/home/ubuntu/venv/snapcraft/lib/python3.5/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/ubuntu/venv/snapcraft/lib/python3.5/site-packages/snapcraft/cli/lifecycle.py", line 192, in snap
    _execute(steps.PRIME, parts=[], pack_project=True, output=output, **kwargs)
  File "/home/ubuntu/venv/snapcraft/lib/python3.5/site-packages/snapcraft/cli/lifecycle.py", line 81, in _execute
    lifecycle.execute(step, project_config, parts)
  File "/home/ubuntu/venv/snapcraft/lib/python3.5/site-packages/snapcraft/internal/lifecycle/_runner.py", line 90, in execute
    executor.run(step, part_names)
  File "/home/ubuntu/venv/snapcraft/lib/python3.5/site-packages/snapcraft/internal/lifecycle/_runner.py", line 194, in run
    self._handle_step(part_names, part, step, current_step, cli_config)
  File "/home/ubuntu/venv/snapcraft/lib/python3.5/site-packages/snapcraft/internal/lifecycle/_runner.py", line 208, in _handle_step
    getattr(self, "_run_{}".format(current_step.name))(part)
  File "/home/ubuntu/venv/snapcraft/lib/python3.5/site-packages/snapcraft/internal/lifecycle/_runner.py", line 250, in _run_pull
    self._run_step(step=steps.PULL, part=part, progress="Pulling")
  File "/home/ubuntu/venv/snapcraft/lib/python3.5/site-packages/snapcraft/internal/lifecycle/_runner.py", line 327, in _run_step
    getattr(part, step.name)()
  File "/home/ubuntu/venv/snapcraft/lib/python3.5/site-packages/snapcraft/internal/pluginhandler/__init__.py", line 415, in pull
    self._runner.pull()
  File "/home/ubuntu/venv/snapcraft/lib/python3.5/site-packages/snapcraft/internal/pluginhandler/_runner.py", line 83, in pull
    "override-pull", self._override_pull_scriptlet, self._sourcedir
  File "/home/ubuntu/venv/snapcraft/lib/python3.5/site-packages/snapcraft/internal/pluginhandler/_runner.py", line 162, in _run_scriptlet
    scriptlet_name, function_call.strip()
  File "/home/ubuntu/venv/snapcraft/lib/python3.5/site-packages/snapcraft/internal/pluginhandler/_runner.py", line 218, in _handle_builtin_function
    function(**function_args)
  File "/home/ubuntu/venv/snapcraft/lib/python3.5/site-packages/snapcraft/internal/pluginhandler/__init__.py", line 434, in _do_pull
    self.source_handler.pull()
  File "/home/ubuntu/venv/snapcraft/lib/python3.5/site-packages/snapcraft/internal/sources/_local.py", line 40, in pull
    copy_function=self.copy_function,
  File "/home/ubuntu/venv/snapcraft/lib/python3.5/site-packages/snapcraft/file_utils.py", line 200, in link_or_copy_tree
    raise NotADirectoryError("{!r} is not a directory".format(source_tree))
NotADirectoryError: '/home/ubuntu/source/make-hello/s' is not a directory
```

Reply:
I think we should refactor this to, in the case of no TTY, only involve Sentry when SNAPCRAFT_ENABLE_SILENT_REPORT=y and otherwise just print the manual reporting (Launchpad) message with the trace.

Reply:
@sergiusens the ordering is due to the traceback being on stderr but the other messages being on stdout, ssh is doing something odd there. Regardless, they should all be printed on the same fd anyway.

Reply:
Fixed. @evandandrea you're right, but that's a large refactor, good for a follow-up.